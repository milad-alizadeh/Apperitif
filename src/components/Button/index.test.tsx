import { fireEvent, render } from '@testing-library/react-native'
import React from 'react'
import { Button, ButtonProps } from './index'

describe('Button', () => {
  it('renders correctly', () => {
    const { getByText } = render(<Button label="Click me" />)
    expect(getByText('Click me')).toBeTruthy()
  })
  it('renders loading indicator when loading is true', () => {
    const { getByTestId } = render(<Button label="Click me" loading />)
    expect(getByTestId('loading-indicator')).toBeTruthy()
  })
  it('calls onPress function when button is pressed', () => {
    const onPress = jest.fn()
    const { getByText } = render(<Button label="Click me" onPress={onPress} />)
    fireEvent.press(getByText('Click me'))
    expect(onPress).toHaveBeenCalled()
  })
})
