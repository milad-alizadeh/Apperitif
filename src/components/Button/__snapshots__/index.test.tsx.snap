// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#DB2031",
      "borderBottomColor": "#DB2031",
      "borderBottomLeftRadius": 12,
      "borderBottomRightRadius": 12,
      "borderBottomWidth": 2,
      "borderLeftColor": "#DB2031",
      "borderLeftWidth": 2,
      "borderRightColor": "#DB2031",
      "borderRightWidth": 2,
      "borderTopColor": "#DB2031",
      "borderTopLeftRadius": 12,
      "borderTopRightRadius": 12,
      "borderTopWidth": 2,
      "height": 48,
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 8,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 8,
    }
  }
>
  <Text
    style={
      [
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        {
          "fontWeight": "500",
        },
        {
          "color": "#fff",
        },
        {
          "fontSize": 20,
          "lineHeight": 28,
        },
      ]
    }
  >
    Click me
  </Text>
</View>
`;
